src/Common/src/TypeSystem/Common/TargetDetails.cs

```diff
@@ -115,11 +115,28 @@ public bool SupportsRelativePointers
         /// <summary>
         /// Gets the maximum alignment to which something can be aligned
         /// </summary>
-        public static int MaximumAlignment
+        public int MaximumAlignment
         {
             get
             {
-                return 8;
+                if (Abi == TargetAbi.ProjectN)
+                {
+                    // ProjectN doesn't support hardware intrinsics
+                    return 8;
```

Comment:
Bumping this value up has negative effects on universal shared code (RyuJIT doesn't concern itself with it, but it's a thing on Project N).

This constant is used in cases like this:

```csharp
class Foo<T>
{
    T myField;
}
```

When T is `__UniversalCanon` (could be a ref type, could be a struct, we have no clue): with maximum supported alignment of 8, we can still assign an offset to `myField` because the field is already aligned at the maximum possible alignment (the previous type ends at offset 8), no matter what T ends up being at runtime.

Once this is bumped to 32, T could be something that requires alignment of 16 or 32 and we can no longer give it an offset (so accessing the field from generated code becomes: go call a helper to find offset of myField). This is a net perf loss. No point in paying it when the backend is UTC.

Reply:
Will ProjectN never support hardware intrinsics? Because it will be a breaking change to modify this in the future if users have a `T` that is a `Vector128<T>` otherwise (the layout of the struct will change, which will impact unsafe code, offsets, etc).

Reply:
> This constant is used in cases like this:

```
class Foo<T>
{
    T myField;
}
```

What happens in CoreCLR in cases like these today? 

For reference types, `Vector64/128/256` alignment requirements should be irrelevant  since the GC will allocate this whole type with an arbitrary offsets anyway.

For struct types, where does this alignment matter? Is it just that the offsets are aligned; or does the JIT actually allocate these types on aligned addresses on the stack?

Reply:
> What happens in CoreCLR in cases like these today?

CoreCLR appropriately packs `T` according to the packing for `T`. We have tests covering this here: https://github.com/dotnet/coreclr/blob/master/tests/src/Interop/StructPacking/StructPacking.cs#L27

Reply:
> For struct types, where does this alignment matter? Is it just that the offsets are aligned; or does the JIT actually allocate these types on aligned addresses on the stack?

The alignment technically matters for all cases. I believe that, for `Vector128<T>` we generally work (since stack alignment defaults to 16), but I don't think we have anything special that guarantees this for stack allocations today (we probably should for both this and RVA statics, since those should be possible to make "pay to play").